packages/config/src/Config.ts

```diff
@@ -473,10 +473,12 @@ function ensureConfigHasDefaultValues({
   const expWithDefaults = { ...exp, name, slug, version, description };
 
   let sdkVersion;
-  try {
-    sdkVersion = getExpoSDKVersion(projectRoot, expWithDefaults);
-  } catch (error) {
-    if (!skipSDKVersionRequirement) throw error;
+  if (!exp.runtimeVersion) {
+    try {
+      sdkVersion = getExpoSDKVersion(projectRoot, expWithDefaults);
+    } catch (error) {
+      if (!skipSDKVersionRequirement) throw error;
+    }
   }
```

Comment:
We should always perform the SDK version check, if you don't want it then pass `skipSDKVersionRequirement` to `getConfig`.